libs/remix-ui/solidity-compiler/src/lib/contract-selection.tsx

```diff
@@ -186,6 +196,48 @@ export const ContractSelection = (props: ContractSelectionProps) => {
     return copyContractProperty('bytecode')
   }
 
+  /**
+   * Take AST and generates a UML diagram of compiled contract as svg
+   * @returns void
+   */
+  const generateUML = () => {
+    try {
+      const currentFile = api.currentFile
+      const ast = content4AST.length > 1 ? parser.parse(content4AST) : parser.parse(api.getCompilationResult().source.sources[currentFile].content)
+      const payload = vizRenderStringSync(convertUmlClasses2Dot(convertAST2UmlClasses(ast, currentFile)))
+      setSVGPayload(payload)
+      setShowViewer(!showViewer)
+    } catch (error) {
+      console.log({ error })
+    }
+  }
+
+  /**
+   * Takes currently compiled contract that has a bunch of imports at the top
+   * and flattens them ready for UML creation. Takes the flattened result
+   * and assigns to a local property
+   * @returns void
+   */
+  const flattenContract = () => {
+    const filePath = api.getCompilationResult().source.target
+    const ast = api.getCompilationResult().data.sources
+    const dependencyGraph = getDependencyGraph(ast, filePath)
+    const sorted = dependencyGraph.isEmpty()
+        ? [filePath]
+        : dependencyGraph.sort().reverse()
+    const sources = api.getCompilationResult().source.sources
+    setContent4AST(concatSourceFiles(sorted, sources))
```

Comment:
flatten the file should create a new file with the suffix `_flattened.sol`

Reply:
Ok. Done

Reply:
did you push?

Reply:
Yes I did